openfisca_core/simulations.py

```diff
@@ -49,17 +60,52 @@ def __init__(self, debug = False, debug_all = False, period = None, tax_benefit_
         self.compact_legislation_by_instant_cache = {}
         self.reference_compact_legislation_by_instant_cache = {}
 
-        self.instantiate_entities()
+        self.instantiate_entities(simulation_json)
+
+    def instantiate_entities(self, simulation_json):
+        if simulation_json:
+            check_type(simulation_json, dict, ['error'])
+            copied_simulation_json = simulation_json.copy()  # Avoid mutating the input
+
+            persons_json = copied_simulation_json.pop(self.tax_benefit_system.person_entity.plural, None)
+
+            if not persons_json:
+                raise SituationParsingError([self.tax_benefit_system.person_entity.plural],
+                    'No person found. At least one person must be defined to run a simulation.')
+            self.persons = self.tax_benefit_system.person_entity(self, persons_json)
+        else:
+            self.persons = self.tax_benefit_system.person_entity(self)
 
-    def instantiate_entities(self):
-        self.persons = self.tax_benefit_system.person_entity(self)
-        setattr(self, self.persons.key, self.persons)
         self.entities = {self.persons.key: self.persons}
+        setattr(self, self.persons.key, self.persons)  # create shortcut simulation.person (for instance)
+
+        for entity_class in self.tax_benefit_system.group_entities:
+            if simulation_json:
+                entities_json = copied_simulation_json.pop(entity_class.plural, {})
+                entities = entity_class(self, entities_json)
+            else:
+                entities = entity_class(self)
+            self.entities[entity_class.key] = entities
```

Comment:
[Normally](https://stackoverflow.com/questions/9056957/correct-way-to-define-class-variables-in-python), what you declare on the first lines of a class are **class attributes** (similar to static attributes in Java).

Using class attributes to define instances attributes default values _technically_ works (only because the default attributes are not mutable), but it's not a good practise. 

I know, we do this all over the codebase 😞